File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int __DFTHelp(TreeNode* curNode, int depth){
        if (curNode == nullptr){
            return depth;
        } else {
            int depthL = __DFTHelp(curNode->left, depth+1);
            int depthR = __DFTHelp(curNode->right, depth+1);
            return max(depthL, depthR);
        }
    }


    int maxDepth(TreeNode* root) {
        return __DFTHelp(root, 0);
    }
};

File: binary-tree-dfs/0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right



class Solution:

    def __DFTHelp(self, curNode, depth):
        if curNode is None:
            return depth
        else:
            depthL = self.__DFTHelp(curNode.left, depth+1)
            depthR = self.__DFTHelp(curNode.right, depth+1)
            return max(depthL, depthR)

    def maxDepth(self, root: Optional[TreeNode]) -> int:
        depth = 0
        return self.__DFTHelp(root, depth)